runtime/src/deterministic.rs

```diff
@@ -46,6 +48,49 @@ use std::{
 };
 use tracing::trace;
 
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct BandwidthLabels {
+    socket: SocketAddr,
+    direction: String,
+}
+
+impl EncodeLabelSet for BandwidthLabels {
```

Comment:
I think you can derive this using `EncodeLabelSet`: https://github.com/commonwarexyz/monorepo/blob/e19234c7b35db18f5d54a9c08096a8a63b0f514c/p2p/src/authenticated/metrics.rs#L20?

I would recommend string-ifying the SocketAddrs when creating this struct (in case that prevents derivation). If not, then feel free to leave as is.